core/src/main/java/org/mule/runtime/core/privileged/exception/TemplateOnErrorHandler.java

```diff
@@ -315,4 +318,17 @@ public void setRootContainerName(String rootContainerName) {
     updateRootContainerName(rootContainerName, this);
   }
 
+  protected void setFlowLocation(Location location) {
+    this.flowLocation = ofNullable(location);
+  }
+
+  /**
+   * Creates a copy of this ErrorHandler, with the defined location to retrieve the {@link ProcessingStrategy}.
+   * This is intended to be used when having references to Global ErrorHandlers, since each instance reference
+   * should run with the processing strategy defined by the flow referencing it
+   * @param localizeFor
+   * @return copy of this ErrorHandler with location to retrieve {@link ProcessingStrategy}
+   */
+  public abstract TemplateOnErrorHandler copy(Location localizeFor);
```

Comment:
rename to something that gives . a better understanding of what is this for (`locateFor`? `buildFor`?)